src/sudoku.rs

```diff
@@ -36,6 +36,26 @@ impl<const N: usize> Display for Row<N> {
     }
 }
 
+#[derive(Debug, Default)]
+struct Scorer {
+    seen: u64,
+    score: u8,
+}
+
+impl Scorer {
+    fn check(&mut self, digit: u8) {
+        let bit = 1 << digit;
```

Comment:
This is a pretty clever idea of using bitwise operations to count the numbers already seen! I'll write some unit tests for this to understand it better :+1: